AI-InvestiBot/implementation.py

```diff
@@ -25,15 +25,38 @@
 from resource_manager import ResourceManager
 
 import numpy as np
+import json
 #from trading_funcs import company_symbols, is_floats
+
 company_symbols = ["AAPL", "HD", "DIS", "GOOG"]
 
-# API keys from alpaca
-YOUR_API_KEY_ID = None
-YOUR_SECRET_KEY = None
-# API keys from AWS lambda, see boto3 documentation
-BUCKET_NAME = 'your_s3_bucket_name'
-OBJECT_KEY = 'your_s3_object_key'
+try: # import keys from config file
+    with open("secrets.config","rb") as f:
+        secrets = json.load(f)
+except FileNotFoundError:
+    print("missing 'secrets.config' file, rename the 'secrets_example.config' file")
+    raise
+
+try:
+    # API keys from alpaca
+    YOUR_API_KEY_ID = secrets["alpaca_api_key"]
+    YOUR_SECRET_KEY = secrets["alpaca_secret_key"]
+    # API keys from AWS lambda, see boto3 documentation
+    BUCKET_NAME = secrets["aws_bucket_name"]
+    OBJECT_KEY = secrets["aws_object_key"]
+except KeyError:
+    print("missing keys in 'secrets.config' file.")
+    raise
+
+if YOUR_API_KEY_ID is None or YOUR_API_KEY_ID=="":
+    raise ValueError("Set your API key ID from alpaca")
+if YOUR_SECRET_KEY is None or YOUR_SECRET_KEY=="":
+    raise ValueError("Set your secret key from alpaca")
+if BUCKET_NAME is None or BUCKET_NAME=="":
+    raise ValueError("Set your BUCKET_NAME from AWS")
+if OBJECT_KEY is None or OBJECT_KEY=="":
+    raise ValueError("Set your OBJECT_KEY from AWS")
```

Comment:
Not all people will want to do both aws.

Reply:
True... But currently the code uses boto3 and isn't boto3 from AWS?
Personally I hate AWS, but this project seems to heavily rely on it...

Takes forever to configure the AWS console correctly and I'm still troubleshooting it.

For json storage I prefer mongodb, atlas has a free tier up to 500mb, it's basically a "dictionary on the cloud"

I have no alternatives for AWS lambda tho, other than my own servers (which has very little compute)

Reply:
Not the project does not rely on aws, was is just an experimental way of running the bot without a laptop open. Sure we can switch it to something else.

P.S: If you do not want the lambda version, run the loop implementation.
There are 2 implementations: the aws, and the loop(just a while True loop).

Reply:
mongodb might be cheaper as well.

Reply:
Also, I can easily add more implementations